### PR TITLE
update dependencies to make it work on Node 18, fix Phaser usage for the new version

### PR DESCRIPTION
**Pull request by Boucaa**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ dist/
 *.kra
 *.kra~
 *.png~
+.idea
```
Comment (vvolhejn) [inline .gitignore:7]: Maybe we could use something more granular? https://stackoverflow.com/questions/11968531/what-to-gitignore-from-the-idea-folder

I'm using VSCode now but there might be other IntelliJ-ers

File: package.json [hunk 2]
```diff
@@ -18,12 +18,18 @@
   },
   "homepage": "https://github.com/allemansratten/levy-sluchat#readme",
```
Comment (vvolhejn) [inline package.json]: Could you fix this while you're at it?

File: src/bullet.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export class Bullet extends Phaser.GameObjects.Container {
     damage: integer
 
     bulletImage: Phaser.GameObjects.Image
-    particles: Phaser.GameObjects.Particles.ParticleEmitterManager
+    // particles: Phaser.GameObjects.Particles.ParticleEmitterManager
```
Comment (vvolhejn) [inline src/bullet.ts]: Leftover comment

File: src/enemy.ts [hunk 4]
```diff
@@ -12,8 +13,8 @@ export abstract class EnemyBase extends Phaser.GameObjects.Sprite {
 
     follower: any
     hp: integer
-    yOffset: number = Phaser.Math.RND.integerInRange(-20, 20);
-    xOffset: number = Phaser.Math.RND.integerInRange(-20, 20);
+    yOffset: number = PMath.RND.integerInRange(-20, 20);
```
Comment (vvolhejn) [inline src/enemy.ts:16]: Did `Phaser.Math` stop working?

File: src/enemy.ts [hunk 5]
```diff
@@ -23,12 +24,12 @@ export abstract class EnemyBase extends Phaser.GameObjects.Sprite {
...
         this.config = stats
         this.stats = {...stats}  // shallow copy, idk how to do deep copy
-        this.follower = { t: 0, vec: new Phaser.Math.Vector2() };
+        this.follower = { t: 0, vec: new PMath.Vector2() };
 
         this.setInteractive();
         this.on('pointerover', () => {
             if (this.scene.input.enabled)
-                (this.scene.scene.get('hudScene') as HudScene).setDescriptionEnemy(this);
+                (this.scene.scene.get<HudScene>('hudScene') as HudScene).setDescriptionEnemy(this);
```
Comment (vvolhejn) [inline src/enemy.ts]: Do we still need `as HudScene` with the parametrized `get()`? Also in other places

File: src/towers.ts [hunk 6]
```diff
@@ -30,13 +29,18 @@ function getEnemy(x, y, range, enemies, numToGet): EnemyBase[] {
     }
     return null
 }
+class Foo extends Phaser.GameObjects.Container {
```
Comment (vvolhejn) [inline src/towers.ts]: This doesn't seem to belong here ;)

File: src/towers.ts [hunk 7]
```diff
@@ -30,13 +29,18 @@ function getEnemy(x, y, range, enemies, numToGet): EnemyBase[] {
...
+    }
+}
 
 
 export class Tower extends Phaser.GameObjects.Container {
     config: TowerConfig
     stats: TowerConfig
 
-    scene: TDScene
+    // scene: TDScene
```
Comment (vvolhejn) [inline src/towers.ts]: Leftover comment

File: src/towers.ts [hunk 8]
```diff
@@ -122,13 +128,13 @@ export class Tower extends Phaser.GameObjects.Container {
             });
             this.rangeIndicator.setVisible(true);
 
-            let hudScene = this.scene.scene.get("hudScene") as HudScene
+            let hudScene = this.scene.scene.get<HudScene>("hudScene") as HudScene
```
Comment (vvolhejn) [inline src/towers.ts]: See comment above

File: src/towers.ts [hunk 9]
```diff
@@ -300,21 +305,21 @@ export class MultishotTurret extends _TowerTurret {
...
                     this.x, this.y, angle,
                     damage,
                     this.parent.stats.range(this.parent.level),
                     this.parent.stats.bulletSpeedMod
                 );
                 this.fireAnimation(angle, damage)
-                if (this.scene == this.scene.metaScene.activeScene)
-                    this.scene.metaScene.soundManager.multishotSound.play();  // this should be in config if this were done properly
+                if (this.tdScene == this.tdScene.metaScene.activeScene)
+                this.tdScene.metaScene.soundManager.multishotSound.play();  // this should be in config if this were done properly
```
Comment (vvolhejn) [inline src/towers.ts]: Weird formatting, no? Missing some indentation

File: webpack.config.ts [hunk 10]
```diff
@@ -26,43 +28,48 @@ module.exports = {
...
+    })
   ],
 
   optimization: {
     splitChunks: {
       cacheGroups: {
         commons: {
           test: /[\\/]node_modules[\\/]/,
-          name: 'vendors',
+          // name: 'vendors',
```
Comment (vvolhejn) [inline webpack.config.ts]: Leftover comment

**Review by vvolhejn — CHANGES_REQUESTED** [review]
Thanks a lot for doing this cleanup! There's still a few leftovers

File: .gitignore (hunk 1 above)
Comment (Boucaa) [reply .gitignore:7]: I used to try this approach for a few years, but once every 2 months they come up with a new garbage file (and a new backwards-incompatible IDE version), so this is just the easy solution. Some files like run configuration might be a good idea to keep, but it's always a discussion about what we want to share and keep personalizaed for each user. So I'd keep everything ignored until we find something that we know we want to share

File: src/enemy.ts (hunk 4 above)
Comment (Boucaa) [reply src/enemy.ts:16]: yeah, it would just break in runtime, no idea why this works though

File: package.json (hunk 2 above)
Comment (Boucaa) [reply package.json]: done

File: src/bullet.ts (hunk 3 above)
Comment (Boucaa) [reply src/bullet.ts]: fixed

File: src/enemy.ts (hunk 5 above)
Comment (Boucaa) [reply src/enemy.ts]: nope, fixed

File: src/towers.ts (hunk 6 above)
Comment (Boucaa) [reply src/towers.ts]: my bad, fixed

File: src/towers.ts (hunk 7 above)
Comment (Boucaa) [reply src/towers.ts]: fixed

File: src/towers.ts (hunk 8 above)
Comment (Boucaa) [reply src/towers.ts]: fixed

File: src/towers.ts (hunk 9 above)
Comment (Boucaa) [reply src/towers.ts]: yeah, idk why, LF eslint wizard :), fixed

File: webpack.config.ts (hunk 10 above)
Comment (Boucaa) [reply webpack.config.ts]: fixed

File: .gitignore (hunk 1 above)
Comment (vvolhejn) [reply .gitignore:7]: Makes sense!

**Review by vvolhejn — APPROVED** [review]
✅

**Pull request merged**